src/OnePlusBot/Base/CommandHandler.cs

```diff
@@ -32,11 +32,60 @@ public CommandHandler(DiscordSocketClient bot, CommandService commands, IService
         public async Task InstallCommandsAsync()
         {
             _bot.MessageReceived += OnCommandReceived;
-            _bot.MessageReceived += OnMessageReceived;  
+            _bot.MessageReceived += OnMessageReceived;
+            _bot.MessageDeleted += OnMessageRemoved;
+            _bot.MessageUpdated += OnMessageUpdatedAsync;
             _commands.CommandExecuted += OnCommandExecutedAsync;
             await _commands.AddModulesAsync(Assembly.GetEntryAssembly(), _services);
         }
 
+        private async Task OnMessageUpdatedAsync(Cacheable<IMessage, ulong> cacheable, SocketMessage message, ISocketMessageChannel socketChannel)
+        {
+            var channel = socketChannel as SocketTextChannel;
```

Comment:
You don't check for null, using as is pointless; Replace with direct cast or `if (socketChannel is SocketTextChannel channel)`